sample-app/src/components/InputDropdown.tsx

```diff
@@ -52,74 +47,120 @@ export default function InputDropdown({
   placeholder,
   screenReaderInstructions,
   screenReaderInstructionsId,
-  options,
-  optionIdPrefix,
+  screenReaderText,
+  children,
   onSubmit = () => {},
-  updateInputValue,
-  updateDropdown,
+  onInputChange,
+  onInputFocus,
   renderButtons = () => null,
   cssClasses
-}: Props): JSX.Element | null {
+}: React.PropsWithChildren<Props>): JSX.Element | null {
   const [{
-    focusedOptionIndex,
+    focusedSectionIndex,
     shouldDisplayDropdown,
   }, dispatch] = useReducer(reducer, {
-    focusedOptionIndex: undefined,
+    focusedSectionIndex: undefined,
     shouldDisplayDropdown: false,
   });
-  const focusOptionId = focusedOptionIndex === undefined
-    ? undefined
-    : `${optionIdPrefix}-${focusedOptionIndex}`;
 
+  const [focusedOptionId, setFocusedOptionId] = useState<string | undefined>(undefined);
   const [latestUserInput, setLatestUserInput] = useState(inputValue);
   const [screenReaderKey, setScreenReaderKey] = useState(0);
 
   const inputRef = useRef<HTMLInputElement>(document.createElement('input'));
+  const dropdownRef = useRef<HTMLDivElement>(document.createElement('div'));
 
   if (!shouldDisplayDropdown && screenReaderKey) {
     setScreenReaderKey(0);
   }
 
+  const childrenArray = React.Children.toArray(children);
+  const childrenWithProps = childrenArray.map((child, index) => {
+    if (!React.isValidElement(child)) {
+      return child;
+    }
+    
+    const modifiedOnClickOption : typeof child.props.onClickOption = (option: typeof child.props.options[0]) => {
```

Comment:
is this `typeof` working? I'd be surprised if it works if the props type is `any`

Reply:
no, it's not working here, so I can remove it and address the typing later with the type guarding you mentioned. but, if we want to avoid having `(option: any)` in this PR, then I could also do the type guarding now